linuxPackages_zen.kernel: fails to build since update to 5.18.0-zen1

The Zen kernel is failing to build for me since the update to 5.18, and so I thought I'd open an issue since I can't seem to find any existing info about it.

**Bisected to https://github.com/NixOS/nixpkgs/commit/208fd4f17325cf506eea151e3614ae2081fa8b6c.**

### Steps To Reproduce

```sh
$ nix build github:NixOS/nixpkgs/208fd4f17325cf506eea151e3614ae2081fa8b6c#linuxPackages_zen.kernel -L
```

### Build Log

```
../drivers/i2c/busses/i2c-nct6775.c: In function 'nct6775_access':
../drivers/i2c/busses/i2c-nct6775.c:221:39: error: this statement may fall through [-Werror=implicit-fallthrough=]
  221 |                         tmp_data.byte = data->byte;
      |                         ~~~~~~~~~~~~~~^~~~~~~~~~~~
../drivers/i2c/busses/i2c-nct6775.c:222:17: note: here
  222 |                 case I2C_SMBUS_BYTE:
      |                 ^~~~
cc1: all warnings being treated as errors
make[3]: *** [../scripts/Makefile.build:288: drivers/i2c/busses/i2c-nct6775.o] Error 1
make[2]: *** [../scripts/Makefile.build:550: drivers/i2c/busses] Error 2
make[1]: *** [../scripts/Makefile.build:550: drivers/i2c] Error 2
make[1]: *** Waiting for unfinished jobs....
```

### Notify maintainers

@Atemu @andresilva 

### Metadata

- system: `"x86_64-linux"`
 - host os: `Linux 5.17.7-zen1, NixOS, 22.05 (Quokka), 22.05.20220518.48037fd`
 - multi-user?: `yes`
 - sandbox: `yes`
 - version: `nix-env (Nix) 2.8.1`
 - channels(root): `""`
 - channels(alex): `""`
 - nixpkgs: `/run/current-system/nixpkgs`

--- a/pkgs/os-specific/linux/kernel/linux-zen.nix
+++ b/pkgs/os-specific/linux/kernel/linux-zen.nix
@@ -2,7 +2,7 @@
 
 let
   # having the full version string here makes it easier to update
-  modDirVersion = "5.18.0-zen1";
+  modDirVersion = "5.18.1-zen1";
   parts = lib.splitString "-" modDirVersion;
   version = lib.elemAt parts 0;
   suffix = lib.elemAt parts 1;
@@ -20,11 +20,14 @@ buildLinux (args // {
     owner = "zen-kernel";
     repo = "zen-kernel";
     inherit rev;
-    sha256 = "sha256-A0QrY1REbRODnHtmyNqVaiLhDgYCECevfHZCxtoQ9kU=";
+    sha256 = "sha256-LCLfLE85NifuskYl2dxLOJEsUNHLegF8ecYyU4xOCtY=";
   };
 
   structuredExtraConfig = with lib.kernel; {
     ZEN_INTERACTIVE = yes;
+    # TODO: Remove once #175433 reaches master
+    # https://nixpk.gs/pr-tracker.html?pr=175433
+    WERROR = no;
   };
 
   extraMeta = {

